src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -192,14 +202,25 @@ public class StoreService : IStoreService
 
         try
         {
+
+            if (_enableDropInvalidDicomJsonMetadata)
+            {
+                // drop invalid metadata
+                foreach (DicomTag tag in storeValidatorResult.InvalidTags)
+                {
+                    dicomDataset.Remove(tag);
+                }
+            }
+
             // Store the instance.
             long length = await _storeOrchestrator.StoreDicomInstanceEntryAsync(
                 dicomInstanceEntry,
+                dicomDataset, // this dataset has potentially been modified for metadata storing
```

Comment:
We shouldn't need this extra param - since DicomDatasets are a reference type, any changes made in this context will be reflected in the object held by `dicomInstanceEntry`.